File: Subclass/subBar_blk.m
//
//  subBar_blk.m
//  BeFit
//
//  Created by Jon Brown on 10/13/14.
//
//

#import "subBar_blk.h"

@implementation subBar_blk

- (void)setLineColor:(NSColor *)color {
    if (color != lineColor) {
        lineColor = [NSColor blackColor];
        [self setNeedsDisplay:YES]; /// We changed what we'd draw, invalidate our drawing.
    }
}

- (void)mouseDown:(NSEvent *)mouseDown {
    clickCount = (clickCount == 6) ? 0 : (clickCount + 1);
    CGFloat hue = clickCount / 6.0;
    [self setLineColor:[NSColor colorWithCalibratedHue:hue saturation:1.0 brightness:1.0 alpha:1.0]];
}

- (void)drawRect:(NSRect)dirtyRect {
    NSRect wholeRect = [self bounds];
    NSColor *myColor = [NSColor colorWithCalibratedRed:0.42 green:0.42 blue:0.42 alpha:1.0f];
    
    NSBezierPath *line = [NSBezierPath bezierPath];
    [line moveToPoint:NSMakePoint(0, wholeRect.size.height)];
    [line lineToPoint:NSMakePoint(wholeRect.size.width, wholeRect.size.height)];
    [line setLineWidth:1.0]; /// Make it easy to see
    [[NSColor blackColor] set]; /// Make future drawing the color of lineColor.
    [line stroke];
}

@end

File: Subclass/SubBarBlk.swift
//
//  SubBarBlk.swift
//  BeFit
//
//  Created by Andrej Slegl on 11/6/19.
//

import Foundation
import Cocoa

class SubBarBlk: NSView {
    private var lineColor: NSColor = NSColor.black {
        didSet {
            needsDisplay = true
        }
    }
    
    private var clickCount: Int = 0
    
    override func mouseDown(with event: NSEvent) {
        clickCount = (clickCount + 1) % 7
        //let hue = CGFloat(clickCount) / 6
        //lineColor = NSColor(calibratedHue: hue, saturation: 1, brightness: 1, alpha: 1)
    }
    
    override func draw(_ dirtyRect: NSRect) {
        let wholeRect = bounds
        //let myColor = NSColor(calibratedRed: 0.42, green: 0.42, blue: 0.42, alpha: 1)
        
        let line = NSBezierPath()
        line.move(to: NSPoint(x: 0, y: wholeRect.height))
        line.line(to: NSPoint(x: wholeRect.width, y: wholeRect.height))
        line.lineWidth = 1 // Make it easy to see
        lineColor.set()
        line.stroke()
    }
}
